src/helpers/helpers.ts

```diff
@@ -63,27 +104,31 @@ export interface WikibaseTimeObject {
 
 export type TimeInputValue = string | WikibaseTimeObject
 
-export type TimeFunction = (wikibaseTime: TimeInputValue) => any
+type TimeFunction<T> = (wikibaseTime: TimeInputValue) => T
 
 // Try to parse the date or return the input
-const bestEffort = (fn: TimeFunction) => (value: TimeInputValue) => {
-  try {
-    return fn(value)
-  } catch (err) {
-    // @ts-ignore
-    value = value.time || value
-
-    const sign = value[0]
-    // @ts-ignore
-    let [ yearMonthDay, withinDay ] = value.slice(1).split('T')
-    yearMonthDay = yearMonthDay.replace(/-00/g, '-01')
-
-    return `${sign}${yearMonthDay}T${withinDay}`
+function bestEffort<T> (fn: TimeFunction<T>) {
+  return (value: TimeInputValue) => {
+    try {
+      return fn(value)
+    } catch (err) {
+      value = typeof value === 'string' ? value : value.time
+
+      const sign = value[0]
+      let [ yearMonthDay, withinDay ] = value.slice(1).split('T')
+      if (!sign || !yearMonthDay || !withinDay) {
+        throw new Error('TimeInput is invalid: ' + JSON.stringify(value))
+      }
+
+      yearMonthDay = yearMonthDay.replace(/-00/g, '-01')
+
+      return `${sign}${yearMonthDay}T${withinDay}`
+    }
   }
 }
 
-const toEpochTime = (wikibaseTime: TimeInputValue): number => toDateObject(wikibaseTime).getTime()
-const toISOString = (wikibaseTime: TimeInputValue): string => toDateObject(wikibaseTime).toISOString()
+const toEpochTime = (wikibaseTime: TimeInputValue) => toDateObject(wikibaseTime).getTime()
```

Comment:
Nice! Return type is not necessary.

Reply:
I do like to specify return types on functions facing outwards (exported ones) as TypeScript then checks if they are correct or not. In this case its an internal type not leaving the file altogether so having assumed types is sometimes more helpful in my opinion.

The fun side effect of this here is that the `bestEffort` method returns a string as fallback which ends up creating the type `string | number` for `toEpochTime`.

Guess that's the kind of thing that only comes to light when starting to use TypeScript 🙃